StockTrader/src/main/java/com/restResource/StockTrader/controller/SellController.java

```diff
@@ -1,28 +1,33 @@
 package com.restResource.StockTrader.controller;
 
+import com.restResource.StockTrader.entity.Investment;
+import com.restResource.StockTrader.entity.InvestmentId;
 import com.restResource.StockTrader.entity.Quote;
 import com.restResource.StockTrader.entity.PendingSell;
+import com.restResource.StockTrader.repository.InvestmentRepository;
 import com.restResource.StockTrader.repository.SellRepository;
 import com.restResource.StockTrader.service.QuoteService;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.ResponseBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
 
 
 @RestController
+@RequestMapping(value = "/sell")
 public class SellController {
 
     private QuoteService quoteService;
 
     private SellRepository sellRepository;
 
-    public SellController(QuoteService quoteService, SellRepository sellRepository) {
+    private InvestmentRepository investmentRepository;
+
+    public SellController(QuoteService quoteService, SellRepository sellRepository, InvestmentRepository investmentRepository) {
```

Comment:
So how does this constructor work? is it called somewhere else? Ushally we used @Inject to annotate beans, like repositories.

Reply:
I'm using constructor injection here. If there's only one constructor you don't need any annotations.